TELCODOCS-1321 - 4.13 vDU reference config updates (#60679)
openshift/openshift-docs: modules/ztp-sno-siteconfig-config-reference.adoc

a|Set the `installConfigOverrides` field to enable or disable optional components prior to cluster installation.
[NOTE]
====
`NodeTuning` is required for 4.13 and later.
====

|`spec.clusters.clusterLabels`
|Configure cluster labels to correspond to the `bindingRules` field in the `PolicyGenTemplate` CRs that you define.
For example, `policygentemplates/common-ranGen.yaml` applies to all clusters with `common: true` set, `policygentemplates/group-du-sno-ranGen.yaml` applies to all clusters with `group-du-sno: ""` set.

|`spec.clusters.crTemplates.KlusterletAddonConfig`
|Optional. Set `KlusterletAddonConfig` to `KlusterletAddonConfigOverride.yaml to override the default `KlusterletAddonConfig` that is created for the cluster.

|`spec.clusters.nodes.hostName`
|For single-node deployments, define a single host.
For three-node deployments, define three hosts.
For standard deployments, define three hosts with `role: master` and two or more hosts defined with `role: worker`.

|`spec.clusters.nodes.biosConfigRef`
|Optional. Configure required host BIOS settings.

|`spec.clusters.nodes.bmcAddress`
|BMC address that you use to access the host.
Applies to all cluster types.

> {ztp} supports iPXE and virtual media booting by using Redfish or IPMI protocols.

Reviewer: ZTP?

For more information about BMC addressing, see the "Additional resources" section.

|`spec.clusters.nodes.bmcCredentialsName`
|Configure the `bmh-secret` CR that you separately create with the host BMC credentials.
When creating the `bmh-secret` CR, use the same namespace as the `SiteConfig` CR that provisions the host.

|`spec.clusters.nodes.bootMode`
|Set the boot mode for the host to `UEFI`.
The default value is `UEFI`. Use `UEFISecureBoot` to enable secure boot on the host.

|`spec.clusters.nodes.diskPartition`
|Optional. The provided example `diskPartition` is used for the image registry.

|`spec.clusters.nodes.ignitionConfigOverride`
|Optional. Allocate partitions for persist storage used by HTTP transport subscription data for PTP and BMER Operators. Adjust Disk ID and and size to the specific hardware.

|`spec.clusters.nodes.cpuset`
|Configure `cpuset` to match value that you set in the cluster `PerformanceProfile` CR `spec.cpu.reserved` field for workload partitioning.

|`spec.clusters.nodes.nodeNetwork`
|Configure the the network settings for the node.

|`spec.clusters.nodes.nodeNetwork.config.interfaces.ipv6`
|Configure the IPv6 address for the host.

> For {sno} clusters with static IP addresses, the node-specific API and Ingress IPs should be the same.

Reviewer: SNO?

// Module included in the following assemblies:
//
// * scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-sites.adoc

:_content-type: REFERENCE
[id="ztp-sno-siteconfig-config-reference_{context}"]
= {sno-caps} SiteConfig CR installation reference

.SiteConfig CR installation options for {sno} clusters
[cols="1,3", options="header"]
|====
|SiteConfig CR field
|Description

|`spec.cpuPartitioningMode`
|Configure workload partitioning by setting the value for `cpuPartitioningMode` to `AllNodes`.
To complete the configuration, specify the `isolated` and `reserved` CPUs in the `PerformanceProfile` CR.

> You enable workload partitioning during cluster installation only.

Reviewer 2: cpuPartitioningMode is not GA in 4.13. This field needs to be marked as Tech Preview.

Can we reword either line 16 or 18 to make clear that enabling is done by setting cpuPartitioningMode to AllNodes. Maybe line 16 "Enable workload partitioning by setting the value...."

|`metadata.name`
|Set `name` to `assisted-deployment-pull-secret` and create the `assisted-deployment-pull-secret` CR in the same namespace as the `SiteConfig` CR.

|`clusterImageSetNameRef`
|Configure the image set available on the hub cluster.
To see the list of supported versions on your hub cluster, run `oc get clusterimagesets`.

|`installConfigOverrides`

> a|Set the `installConfigOverrides` field to enable or disable optional components prior to cluster installation.

Reviewer 2: The value given in the reference configuration should be used as specified. If they bring additional components back in the system may require additional reserved CPU.

====

|`spec.clusters.clusterLabels`
|Configure cluster labels to correspond to the `bindingRules` field in the `PolicyGenTemplate` CRs that you define.
For example, `policygentemplates/common-ranGen.yaml` applies to all clusters with `common: true` set, `policygentemplates/group-du-sno-ranGen.yaml` applies to all clusters with `group-du-sno: ""` set.

|`spec.clusters.crTemplates.KlusterletAddonConfig`
|Optional. Set `KlusterletAddonConfig` to `KlusterletAddonConfigOverride.yaml to override the default `KlusterletAddonConfig` that is created for the cluster.

|`spec.clusters.nodes.hostName`
|For single-node deployments, define a single host.
For three-node deployments, define three hosts.
For standard deployments, define three hosts with `role: master` and two or more hosts defined with `role: worker`.

|`spec.clusters.nodes.bmcAddress`
|BMC address that you use to access the host. Applies to all cluster types. {ztp} supports iPXE and virtual media booting by using Redfish or IPMI protocols. To use iPXE booting, you must use {rh-rhacm} 2.8 or later. For more information about BMC addressing, see the "Additional resources" section.

|`spec.clusters.nodes.biosConfigRef`

> |Optional. Configure required host BIOS settings.

Reviewer 2: I believe that this is not yet a GA feature. @tliu2021 can you confirm please.

Reviewer 3: Yes, this is not yet a GA feature and it should be removed from the custom doc.

|`spec.clusters.nodes.bmcAddress`
|BMC address that you use to access the host.
Applies to all cluster types.

> {ztp} supports iPXE and virtual media booting by using Redfish or IPMI protocols.

Reviewer 2: In the far edge use case we are only using virtual media.

To use iPXE booting, you must use {rh-rhacm} 2.8 or later.
For more information about BMC addressing, see the "Additional resources" section.

|`spec.clusters.nodes.bmcCredentialsName`
|Configure the `bmh-secret` CR that you separately create with the host BMC credentials.
When creating the `bmh-secret` CR, use the same namespace as the `SiteConfig` CR that provisions the host.

|`spec.clusters.nodes.bootMode`
|Set the boot mode for the host to `UEFI`.
The default value is `UEFI`. Use `UEFISecureBoot` to enable secure boot on the host.

|`spec.clusters.nodes.diskPartition`

> |Optional. The provided example `diskPartition` is used for the image registry.

Reviewer 2: Prefer to not reference an image registry here. This can be simply "is used to configure additional disk partitions"

|`spec.clusters.nodes.ignitionConfigOverride`

> |Optional. Allocate partitions for persist storage used by HTTP transport subscription data for PTP and BMER Operators. Adjust Disk ID and and size to the specific hardware.

Reviewer 2: ```suggestion
|Optional. May be used to allocate partitions for persist storage. Adjust Disk ID and and size to the specific hardware.
```
